feat: Implement assignee filter for team todos

Achintya-Chatterjee opened the pull request:
<!-- Date Here in dd mmm yyyy-->
Date: `10th December, 2025`
<!--Developer Name Here-->
Developer Name: @Achintya-Chatterjee 

---

## Issue Ticket Number
<!--Issue ticket this PR closes-->
- Closes #228 
## Description
<!--Description of the changes made in this PR-->
- Add `assigneeId` filtering support to `GetTaskReqDto`.
- Update `TeamTasks` page to include the new `TeamFilters` component.
- Implement `TeamFilters` component:
  - Uses `DropdownMenu` and `Command` for a searchable assignee list.
  - Supports multi-select for assignees.
  - Styles the filter button with specific red styling (`border-red-500`, `text-red-500`) and prominent border (`border-2`) to match design.
  - Adjusts layout spacing (`-ml-2`) for better visual alignment with the "Include Done" switch.
- Configure global `paramsSerializer` in `apiClient` to correctly serialize array query parameters (e.g., `key=val1&key=val2`) for backend compatibility.
- Simplify `TasksApi.getTasks` to use standard parameter passing.
- Update route validation in `_internal.teams.$teamId.todos` to handle `assigneeId` as an array.
### Documentation Updated?

- [ ] Yes
- [x] No

<!--Additional notes about documentation update if applicable-->

### Under Feature Flag

- [ ] Yes
- [x] No

<!--Indicate if changes are under a feature flag-->

### Database Changes

- [ ] Yes
- [x] No

<!--Notes on any database changes-->

### Breaking Changes

- [ ] Yes
- [x] No

<!--Notes on breaking changes or related issue tickets if applicable-->

### Development Tested?

- [x] Yes
- [ ] No

<!--Confirmation of local testing during development-->

## Screenshots
<details>
<summary>Screenshot 1</summary>
<!-- Attach your screenshots here👇-->

https://github.com/user-attachments/assets/949de307-93a2-46f1-8b09-ef9f106e3448

</details>

<!--Attach or link to relevant screenshots or visual aids-->

## Test Coverage
<details>
<summary>Screenshot 1</summary>

<!-- Attach your screenshots here👇-->

</details>

<!--Attach Details on test coverage and outcomes-->

## Additional Notes

<!--Any additional notes, considerations or explanations for reviewers-->

Achintya-Chatterjee committed:
feat: Implement assignee filter for team todos

Achintya-Chatterjee commented on src/modules/teams/components/team-filters.tsx:
```diff
@@ -0,0 +1,140 @@
+      <DropdownMenuTrigger asChild>
+        <Button
+          variant="outline"
+          className="-ml-2 border-2 border-red-500 text-red-500 hover:border-red-600 hover:bg-red-50 hover:text-red-600"
```
<img width="2946" height="2038" alt="image" src="https://github.com/user-attachments/assets/73d97e8f-0445-42fd-a7bc-368022c37af7" />
This was in the requirement. You can see my attached image to see that the Filter Button has red borders

AnujChhikara replied:
looks very odd with this red color


The red border was specified in the original design requirements. However, I can change to neutral colours if the team prefers.
<img width="2946" height="2038" alt="image" src="https://github.com/user-attachments/assets/72c77144-81e4-4ca6-9be7-d8c14051876b" />

Achintya-Chatterjee committed:
refactor: optimize team data fetching in TeamFilters

- Remove duplicate team query in TeamFilters component
- Pass team data as prop from parent TeamTasks component
- Update TeamTasks query to include member=true for users list
- Remove unused imports and error handling from TeamFilters
- Eliminate redundant network request and improve data consistency

Achintya-Chatterjee commented on src/modules/teams/team-tasks.tsx:
```diff
@@ -116,6 +117,11 @@ export const TeamTasks = ({ teamId }: TeamTasksProps) => {
+      ? searchParams.assigneeId
+      : searchParams.assigneeId
+        ? [searchParams.assigneeId]
+        : undefined) as string[] | undefined,
```
fixed this, thanks for the suggestion

Achintya-Chatterjee commented on src/modules/teams/components/team-filters.tsx:
```diff
@@ -0,0 +1,140 @@
+    queryFn: () => TeamsApi.getTeamById.fn({ teamId, member: true }),
+  })
+
+  const teamMembers = team?.users ?? []
```
fixed this, thanks for the suggestion.

AnujChhikara commented on src/modules/teams/components/team-filters.tsx:
```diff
@@ -0,0 +1,135 @@
+      : searchParams.assigneeId
+        ? [searchParams.assigneeId]
+        : []
+  ) as string[]
```
no need of as string []

Achintya-Chatterjee replied:
fixed, removed this.

AnujChhikara commented on src/modules/teams/components/team-filters.tsx:
```diff
@@ -0,0 +1,135 @@
+      <DropdownMenuTrigger asChild>
+        <Button
+          variant="outline"
+          className="-ml-2 border-2 border-red-500 text-red-500 hover:border-red-600 hover:bg-red-50 hover:text-red-600"
```
nit: avoid using negative values

AnujChhikara commented on src/modules/teams/components/team-filters.tsx:
```diff
@@ -0,0 +1,135 @@
+    })
+  }
+
+  const handleClearAssignees = () => {
```
are we clearing the assignee filter when we click on the clear button not able to find that logic ??

Achintya-Chatterjee replied:
The logic is in `handleClearAssignees()` at line 58, which sets `assigneeId: undefined` to remove all filters. This is called when clicking the `Clear filters` button at line 99.

AnujChhikara commented on src/modules/teams/components/team-filters.tsx:
```diff
@@ -0,0 +1,135 @@
+        search: prev.search,
+        assigneeId: newAssigneeIds.length ? newAssigneeIds : undefined,
+      }),
+    })
```
how many assignee i can send in the api??

Achintya-Chatterjee replied:
The user can select a single assignee or select multiple assignees that are part of the team.

AnujChhikara commented on src/modules/teams/components/team-filters.tsx:
```diff
@@ -0,0 +1,135 @@
+        status: prev.status,
+        search: prev.search,
+        assigneeId: newAssigneeIds.length ? newAssigneeIds : undefined,
+      }),
```
can we add debounce right now looks like get triggered as soon we select the assignee

Achintya-Chatterjee replied:
Implement the debounced function applied on dropdown close.

AnujChhikara commented on src/routes/_internal.teams.$teamId.todos.tsx:
```diff
@@ -7,6 +7,11 @@ export const Route = createFileRoute('/_internal/teams/$teamId/todos')({
+        ? search.assigneeId
+        : search.assigneeId
+          ? [search.assigneeId]
+          : undefined) as string[] | undefined,
```
I think there is not need of type assertion here

Achintya-Chatterjee replied:
fixed

AnujChhikara commented on src/api/tasks/tasks.types.ts:
```diff
@@ -39,6 +39,7 @@ export type TEditTask = {
 export type GetTaskReqDto = {
   teamId?: string
   status?: string
+  assigneeId?: string[]
```
I think assigneeIds make more sense here

Achintya-Chatterjee replied:
fixed this

AnujChhikara commented:
Overall I don't like the select dropdown approach here going ahead one team can have 500 member or more and the member api we might have to paginate so this implementation would not work and we have to change this.

Achintya-Chatterjee commented:
> Overall I don't like the select dropdown approach here going ahead one team can have 500 member or more and the member api we might have to paginate so this implementation would not work and we have to change this.

I think this is not needed yet; will revisit if the product asks for it. Also, the API already limits results to team members, and we can add a limit when there’s a real need.

Achintya-Chatterjee committed:
refactor: address PR feedback for assignee filter

- Remove unnecessary type assertions for improved type safety
- Implement debounced filter application (applies on dropdown close)
- Adjust button styling to match search bar height (border: 1px)
- Optimize spacing with -ml-3 for visual balance
- Maintain red button color per original design requirements

Addresses feedback from PR review regarding code quality,
UX improvements, and styling consistency.
